File: c++/usaco/bronze/square.cpp
#include <bits/stdc++.h>
#define ll long long
using namespace std;

int main() {
  freopen("square.in","r",stdin);
  freopen("square.out","w",stdout);
  int x,y;
  int lx=0,sx=10,ly=0,sy=10;
  for(int i=0;i<4;i++){
    cin>>x>>y;
    lx=max(lx,x);
    ly=max(ly,y);
    sx=min(sx,x);
    sy=min(sy,y);
  }
  int rx=lx-sx,ry=ly-sy,a;
  if(rx>ry)
    a=rx*rx;
  else
    a=ry*ry;
  cout<<a;
  return 0;
}

File: c++/usaco/bronze/square.py
import sys

sys.stdin = open('square.in', 'r')
sys.stdout = open('square.out', 'w')

rect1 = [int(i) for i in input().split()]
rect2 = [int(i) for i in input().split()]

ans_x = max(rect1[2], rect2[2]) - min(rect1[0], rect2[0])
ans_y = max(rect1[3], rect2[3]) - min(rect1[1], rect2[1])

print(max(ans_x, ans_y) ** 2)